nav2_smac_planner/src/smac_planner_hybrid.cpp

```diff
@@ -413,314 +271,117 @@ nav_msgs::msg::Path SmacPlannerHybrid::createPlan(
+    _expansions_publisher.publish(msg);
 
     // plot footprint path planned for debug
-    if (_planned_footprints_publisher->get_subscription_count() > 0) {
-      auto marker_array = std::make_unique<visualization_msgs::msg::MarkerArray>();
-      for (size_t i = 0; i < plan.poses.size(); i++) {
-        const std::vector<geometry_msgs::msg::Point> edge =
-          transformFootprintToEdges(plan.poses[i].pose, _costmap_ros->getRobotFootprint());
-        marker_array->markers.push_back(createMarker(edge, i, _global_frame, _clock->now()));
+    if (_planned_footprints_publisher.getNumSubscribers() > 0) {
+      visualization_msgs::MarkerArray marker_array;
+      for (size_t i = 0; i < output_path.poses.size(); i++) {
+        const std::vector<geometry_msgs::Point> edge =
+          transformFootprintToEdges(output_path.poses[i].pose, _costmap_ros->getRobotFootprint());
+        marker_array.markers.push_back(createMarker(edge, i, _global_frame, ros::Time::now()));
       }
 
-      if (marker_array->markers.empty()) {
-        visualization_msgs::msg::Marker clear_all_marker;
-        clear_all_marker.action = visualization_msgs::msg::Marker::DELETEALL;
-        marker_array->markers.push_back(clear_all_marker);
+      if (marker_array.markers.empty()) {
+        visualization_msgs::Marker clear_all_marker;
+        clear_all_marker.action = visualization_msgs::Marker::DELETEALL;
+        marker_array.markers.push_back(clear_all_marker);
       }
-      _planned_footprints_publisher->publish(std::move(marker_array));
+      _planned_footprints_publisher.publish(marker_array);
     }
   }
 
   // Find how much time we have left to do smoothing
-  steady_clock::time_point b = steady_clock::now();
-  duration<double> time_span = duration_cast<duration<double>>(b - a);
-  double time_remaining = _max_planning_time - static_cast<double>(time_span.count());
+  ros::Time b = ros::Time::now();
+  double time_remaining = _max_planning_time - (b - a).toSec();
 
 #ifdef BENCHMARK_TESTING
   std::cout << "It took " << time_span.count() * 1000 <<
     " milliseconds with " << num_iterations << " iterations." << std::endl;
 #endif
 
-  // Smooth plan
+  // Smooth output_path
   if (_smoother && num_iterations > 1) {
-    _smoother->smooth(plan, costmap, time_remaining);
+    _smoother->smooth(output_path, costmap, time_remaining);
   }
 
 #ifdef BENCHMARK_TESTING
-  steady_clock::time_point c = steady_clock::now();
-  duration<double> time_span2 = duration_cast<duration<double>>(c - b);
-  std::cout << "It took " << time_span2.count() * 1000 <<
+  ros::Time c = ros::Time::now();
+  std::cout << "It took " << (c - b).toSec() * 1000 <<
     " milliseconds to smooth path." << std::endl;
 #endif
 
-  return plan;
+  plan = std::move(output_path.poses);
+  return mbf_msgs::GetPathResult::SUCCESS;
 }
 
-rcl_interfaces::msg::SetParametersResult
-SmacPlannerHybrid::dynamicParametersCallback(std::vector<rclcpp::Parameter> parameters)
```

Comment:
equivalent to our dynamic reconfigure callback, except that I split into planner and path smoother reconfigurations